stdlib/LinearAlgebra/src/LinearAlgebra.jl

```diff
@@ -468,6 +468,10 @@ _evview(S::SymTridiagonal) = @view S.ev[begin:length(S.dv) - 1]
 _zeros(::Type{T}, b::AbstractVector, n::Integer) where {T} = zeros(T, max(length(b), n))
 _zeros(::Type{T}, B::AbstractMatrix, n::Integer) where {T} = zeros(T, max(size(B, 1), n), size(B, 2))
 
+# append a zero element / drop the last element
+_pushzero(A) = (B = similar(A, length(A)+1); B[begin:end-1] .= A; B[end] = zero(eltype(B)); B)
+_droplast(A) = (pop!(A); A)
```

Comment:
```suggestion
```
Why not just call `pop!`? Below, you apply it to freshly allocated memory, so no risk of modifying the diagonals.

Reply:
I am calling `pop!`... The purpose of this function is to return the array instead of the removed element; this makes it easier to use.

Reply:
I messed up the return value of `pop!`. The correct command would be
```suggestion
_droplast!(A) = deleteat!(A, lastindex(A))
```